File: ClassroomServer-Magazine/Models/StudentTrainingGrade.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ClassroomServer.Models
{
    public class StudentTrainingGrade
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public Guid StudentHistoryId { get; set; }

        [Required]
        public Guid TrainingId { get; set; }

        [Required]
        public float Grade { get; set; }

        public List<Guid>? TrainingTasksInfosIds { get; set; } = null;

        public virtual ICollection<StudentTrainingTaskInfo>? TrainingTasksInfos { get; set; } = null;
    }
}

File: ClassroomMgmt-Magazine/src/models/StudentTrainingGrade.ts
export interface StudentTrainingGrade {
    id: string;
    studentHistoryId: string;
    trainingId: string;
    grade: number;
}
export const EMPTY_STUDENT_TRAINING_GRADE: StudentTrainingGrade = {
    id: "",
    studentHistoryId: "",
    trainingId: "",
    grade: 0,
}
